src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        if (!interfaceClazz.equals(ProductServiceImpl.class)
+                && !interfaceClazz.equals(ProductService.class)
+                && !interfaceClazz.equals(ProductParser.class)
+                && !interfaceClazz.equals(ProductParserImpl.class)
+                && !interfaceClazz.equals(FileReaderServiceImpl.class)
+                && !interfaceClazz.equals(FileReaderService.class)) {
```

Comment:
this if statement contains lots of lines which will only grow if we will have more classes in future.

You put this classes in map in findImplementation() method so you can just call findImplementation method to get implementation of your interface that you put in method.

Reply:
make sense. Thanks. reworked